speechbrain/lobes/models/flair/embeddings.py

```diff
@@ -0,0 +1,127 @@
+"""Wrappers for Flair embedding classes
+
+Authors
+* Sylvain de Langen 2024
+"""
+
+import flair
+from flair.data import Sentence
+from flair.embeddings import Embeddings
+from typing import List, Union
+import torch
+
+from speechbrain.utils.fetching import fetch
+
+
+class FlairEmbeddings:
+    """
+    Simple wrapper for generic Flair embeddings.
+
+    Arguments
+    ---------
+    embeddings : Embeddings
+        The Flair embeddings object. If you do not have one initialized, use
+        :meth:`~FlairEmbeddings.from_hf` instead.
+    """
+
+    def __init__(self, embeddings: Embeddings) -> None:
+        self.embeddings = embeddings
+
+    @staticmethod
+    def from_hf(
+        embeddings_class,
+        source,
+        save_path="./model_checkpoints",
+        filename="model.bin",
+        *args,
+        **kwargs,
+    ) -> "FlairEmbeddings":
+        """Fetches and load flair embeddings according to the
+        :func:`speechbrain.utils.fetching.fetch` semantics. Embedding files will
+        be saved into a unique subdirectory in `save_path`.
+
+        Arguments
+        ---------
+        embeddings_class : class
+            The class to use to initialize the model, e.g. `FastTextEmbeddings`.
+        source : str
+            The location of the model (a directory or HF repo, for instance).
+        save_path : str, optional
+            The saving location for the model (i.e. the root for the download or
+            symlink location).
+        filename : str, optional
+            The filename of the model. The default is the usual filename for
+            this kind of model.
+        *args
+            Extra positional arguments to pass to the flair class constructor
+        **kwargs
+            Extra keyword arguments to pass to the flair class constructor
+        """
+
+        target = save_path + "/flair-emb--" + source.replace("/", "--") + "/"
```

Comment:
i suppose that the target template is what flair is using right? Maybe we should leave it as an optional argument? I don't know much about flair but maybe this could make sense. Additionally, I would say you should add a small comment explaining what you are doing